cp23/hpi-fun2/motivation/notes.md

```diff
@@ -0,0 +1,88 @@
+# Motivation
+
+The problem is about finding the path coverage of the tree. 
+
+The solution can be found as follows:
+
+## Optimal Solution
+
+> 1. Find the number of unreachable nodes (nodes, that exceed the allowed path length).
+>    - _This number is the second part of the solution_.
+>    - Also nodes, that have an unreachable parent are themselves unreachable, even if they do not exceed the path length.
+> 2. Find the number of inner nodes and leafs.
+>    - A leaf is a node, that does not have any reachable children.
+>      - That means, either the node is a classical leaf or all children are unreachable.
+>    - An inner node is a node, that is neither reachable nor 
+> 3. The first part of the solution is calculated as follows:
+>    ```
+>    max(1, #(leafs) - #(inner nodes))
```

Comment:
I don't think this is true: [drawing.pdf](https://github.com/compprog-lecture-tools/fun-contest-2023/files/12182628/drawing.pdf) gives a counter-example. We have 3 leafs and 2 inner nodes, max(1,3-2) = 1, but there are two coaches required